test/pylib/scylla_cluster.py

```diff
@@ -32,70 +34,56 @@
 from cassandra.cluster import EXEC_PROFILE_DEFAULT  # pylint: disable=no-name-in-module
 from cassandra.policies import WhiteListRoundRobinPolicy  # type: ignore
 
-#
-# Put all Scylla options in a template file. Sic: if you make a typo in the
-# configuration file, Scylla will boot fine and ignore the setting.
-# Always check the error log after modifying the template.
-#
-SCYLLA_CONF_TEMPLATE = """cluster_name: {cluster_name}
-developer_mode: true
-
-# Allow testing experimental features. Following issue #9467, we need
-# to add here specific experimental features as they are introduced.
-
-enable_user_defined_functions: true
-experimental: true
-experimental_features:
-    - raft
-    - udf
-
-data_file_directories:
-    - {workdir}/data
-commitlog_directory: {workdir}/commitlog
-hints_directory: {workdir}/hints
-view_hints_directory: {workdir}/view_hints
-
-listen_address: {host}
-rpc_address: {host}
-api_address: {host}
-prometheus_address: {host}
-alternator_address: {host}
-
-seed_provider:
-    - class_name: org.apache.cassandra.locator.simple_seed_provider
-      parameters:
-          - seeds: {seeds}
-
-skip_wait_for_gossip_to_settle: 0
-ring_delay_ms: 0
-num_tokens: 16
-flush_schema_tables_after_modification: false
-auto_snapshot: false
-
-# Significantly increase default timeouts to allow running tests
-# on a very slow setup (but without network losses). Note that these
-# are server-side timeouts: The client should also avoid timing out
-# its own requests - for this reason we increase the CQL driver's
-# client-side timeout in conftest.py.
-
-range_request_timeout_in_ms: 300000
-read_request_timeout_in_ms: 300000
-counter_write_request_timeout_in_ms: 300000
-cas_contention_timeout_in_ms: 300000
-truncate_request_timeout_in_ms: 300000
-write_request_timeout_in_ms: 300000
-request_timeout_in_ms: 300000
-
-# Set up authentication in order to allow testing this module
-# and other modules dependent on it: e.g. service levels
-
-authenticator: {authenticator}
-authorizer: {authorizer}
-strict_allow_filtering: true
-
-permissions_update_interval_in_ms: 100
-permissions_validity_in_ms: 100
-"""
+
+def make_scylla_conf(workdir: pathlib.Path, host_addr: str, seed_addrs: List[str], cluster_name: str) -> dict[str, object]:
+    return {
+        'cluster_name': cluster_name,
+        'workdir,W': str(workdir.resolve()),
```

Comment:
I guess this is not a strict prerequisite to committing this patch, but I'd like to understand this "workdir,W" weirdness. The command line option "--workdir" works (I use it every day), so I would have hoped the yaml option "workdir:" also works. A comment in conf/scylla.yaml also suggests that it is supposed to work. If it doesn't, we have some sort of bug in our option parsing or something, and we should open an issue about this, and fix it.

Reply:
https://github.com/scylladb/scylladb/issues/11503